poutyne/framework/callbacks/progress.py

```diff
@@ -75,6 +75,36 @@ def on_train_batch_end(self, batch_number: int, logs: Dict):
             self.color_progress.on_train_batch_end(step_times_rate, batch_number, metrics_str)
             self.last_step = batch_number
 
+    def on_test_begin(self, logs: Dict) -> None:
+        self.step_times_weighted_sum = 0.
+
+        self.metrics = ['loss'] + self.model.metrics_names
+        self.steps = self.params['steps']
+
+        if self.progress_bar and self.steps is not None:
+            self.color_progress.set_progress_bar(self.steps)
+
+    def on_test_batch_end(self, batch_number: int, logs: Dict) -> None:
+        if self.equal_weights:
+            self.step_times_weighted_sum += logs['time']
+            step_times_rate = self.step_times_weighted_sum / batch_number
+        else:
+            self.step_times_weighted_sum += batch_number * logs['time']
+            normalizing_factor = (batch_number * (batch_number + 1)) / 2
+            step_times_rate = self.step_times_weighted_sum / normalizing_factor
```

Comment:
I think this should go in its own method since it seems repeated from `on_train_batch_end`.

Reply:
23d1cb5